scripts/zkai_attention_kv_multihead_quantized_softmax_receipt_gate.py

```
@@ -0,0 +1,800 @@
+        per_head_step_row_counts.append({"head_index": head_index, "step_index": step_index, "rows": len(step_rows)})
+        for row in step_rows:
+            if row["weight_denominator"] != recomputed_denominator:
+                raise MultiheadQuantizedSoftmaxReceiptGateError(f"{profile.profile_id} weight denominator drift")
+            expected_output = outputs[output_index]
+            for dim, (output, remainder, numerator) in enumerate(
+                zip(row["attention_output"], row["output_remainder"], row["weighted_numerator"], strict=True)
+            ):
+                if output != expected_output[dim]:
+                    raise MultiheadQuantizedSoftmaxReceiptGateError(
+                        f"{profile.profile_id} output row/list split-brain drift"
+                    )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)